### PR TITLE
AG-8555 Add visible range scale to axis tick calculation

### PR DESCRIPTION
**Pull request by lsjroberts**
https://ag-grid.atlassian.net/browse/AG-8555

### CODE REVIEW

File: packages/ag-charts-community/src/chart/axis/axis.ts [hunk 1]
```diff
@@ -711,8 +711,6 @@ export abstract class Axis<S extends Scale<D, number, TickInterval<S>> = Scale<a
             return;
         }
 
-        this.setTickCount();
```
Comment (lsjroberts) [inline packages/ag-charts-community/src/chart/axis/axis.ts:714]: This call doesn't do anything as it short circuit returns if no count param or not continuous (skipped on line 711).

File: packages/ag-charts-community/src/chart/axis/axis.ts [hunk 2]
```diff
@@ -1145,14 +1143,21 @@ export abstract class Axis<S extends Scale<D, number, TickInterval<S>> = Scale<a
...
 
         return max - min;
     }
 
+    protected calculateVisibleRange() {
+        const { visibleRange } = this;
+        const visibleScale = 1 / (visibleRange[1] - visibleRange[0]);
+
+        return this.calculateAvailableRange() * visibleScale;
+    }
```
Comment (lsjroberts) [inline packages/ag-charts-community/src/chart/axis/axis.ts:1159]: The `calculateAvailableRange()` function is still used for the label collision checking

File: packages/ag-charts-community/src/chart/axis/axis.ts (hunk 1 above)
Comment (alantreadway) [reply packages/ag-charts-community/src/chart/axis/axis.ts:714]: Nice spot!

**Review by alantreadway — APPROVED** [review]
LGTM

**Pull request merged**